pandas/core/arrays/timedeltas.py

```diff
@@ -217,6 +217,32 @@ def _simple_new(
         result._dtype = TD64NS_DTYPE
         return result
 
+    @classmethod
+    def _from_sequence_strict(
+        cls, data, *, dtype=TD64NS_DTYPE, copy: bool = False
+    ) -> "TimedeltaArray":
+        # GH#37179 eventually we want _from_sequence to be strict
+        if dtype:
+            _validate_td64_dtype(dtype)
+
+        data, copy = dtl.ensure_arraylike(data, copy)
+
+        if data.dtype.kind == "m":
+            pass
+        elif data.dtype == object:
+            inferred = lib.infer_dtype(data)
+            if inferred in ["timedelta64", "timedelta", "empty"]:
+                pass
+            else:
+                raise TypeError(inferred)
+        elif is_string_dtype(data.dtype):
+            # TODO: should go through from_sequence_of_strings?
+            pass
```

Comment:
Shouldn't we also raise here? (strict means for me also no strings, only actual Timestamp/datetime objects)

Reply:
yep, updated